riscv, qemu_fw_cfg: Add support for RISC-V architecture

MahnoKropotkinvich opened the pull request:
https://github.com/RVCK-Project/rvck-olk/issues/106

oervci commented:
开始测试

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/103/consoleFull

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/98/consoleFull

oervci commented:
Lava check fail! log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/rvck-lava-trigger/128/consoleFull

MahnoKropotkinvich commented:
/check

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/99/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/104/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/403

xmzzz commented:
确认下补丁来自主线哪个版本，补丁头需要改一下，
主线补丁 “from” 是必选项，如：`from mainline-6.17`

oervci commented:
开始测试

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/108/consoleFull

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/109/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/113/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/416

xmzzz commented on drivers/firmware/Kconfig:
```diff
@@ -155,7 +155,7 @@ config RASPBERRYPI_FIRMWARE
 config FW_CFG_SYSFS
 	tristate "QEMU fw_cfg device support in sysfs"
-	depends on SYSFS && (ARM || ARM64 || PARISC || PPC_PMAC || SPARC || X86 || SW64)
+	depends on SYSFS && (ARM || ARM64 || PARISC || PPC_PMAC || SPARC || X86 || SW64 || RISCV)
```
这里保持原补丁的首字母排列顺序。顺带给 SW64 也纠正下。

MahnoKropotkinvich replied:
好 请等待下个force push

xmzzz reviewed (changes requested):
补丁描述信息中的换行不对，每行字符过多，检查下您的编辑器。
描述信息内容保持与原补丁一致。

MahnoKropotkinvich committed:
riscv, qemu_fw_cfg: Add support for RISC-V architecture

mainline inclusion
from mainline-Linux 6.7-rc2
commit f2de37a572853d340f945a7748f74e3ed8c6b743
category: feature
bugzilla: #106

--------------------------------

Qemu fw_cfg support was missing for RISC-V, which made it hard to do
proper vmcore dumps from qemu.

Add the missing RISC-V arch-defines.

You can now do vmcore dumps from qemu. Add "-device vmcoreinfo" to the
qemu command-line. From the qemu monitor:
  (qemu) dump-guest-memory vmcore

The vmcore can now be used, e.g., with the "crash" utility.

Acked-by: "Michael S. Tsirkin" <mst@redhat.com>
Acked-by: Alistair Francis <alistair.francis@wdc.com>
Tested-by: Clément Léger <cleger@rivosinc.com>
Signed-off-by: Björn Töpel <bjorn@rivosinc.com>
Message-Id: <20231012102852.234442-1-bjorn@kernel.org>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Signed-off-by: Xuhai Chang <xuhai.oerv@isrc.iscas.ac.cn>

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/110/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/114/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/417
